src/js/D3SVGCellRenderer.js

```diff
@@ -0,0 +1,196 @@
+
+		self.cell_area.selectAll('*').remove();
+		self.svg = self.cell_area.append('svg')
+		.attr('width', (self.oncoprint_config.get('cell_width') + self.oncoprint_config.get('cell_padding'))*self.data.length)
+		.attr('height', self.track_config.get('track_height'));
+
+		self.g = self.svg.selectAll('g').data(self.data, self.track_config.get('datum_id')).enter().append('g').classed('cell', true);
+		self.updateCells();
+	};
+
+	self.updateCells = function() {
+		var id_order = utils.invert_array(self.oncoprint_config.get('id_order'));
+		self.g.transition()
+		.attr('transform', function(d,i) {
+				return utils.translate(id_order[self.track_config.get('datum_id')(d)]*(self.oncoprint_config.get('cell_width') + self.oncoprint_config.get('cell_padding')), 0);
+			});
+
+		self.drawCells();
+	};
+
+	self.drawCells = function() {
+		self.g.selectAll('*').remove();
+		self.rule_set.apply(self.g, self.data, self.track_config.get('datum_id'));
+		self.drawHitZones();
+	};
+
+	self.drawHitZones = function() {
+		self.g.selectAll('rect.hit').remove();
+		var hits = self.g.append('rect').classed('hit', true)
+			.attr('width', self.oncoprint_config.get('cell_width'))
+			.attr('height', self.track_config.get('cell_height'))
+			.attr('stroke', 'rgba(0,0,0,0)')
+			.attr('fill', 'rgba(0,0,0,0)');
+		// bind events
+		var eventData = function(d, i, ctx) {
+			return {datum: d, index: i, g:d3.select(ctx.parentNode)};
+		}
+		hits.on('click', function(d, i){
+			$(self).trigger('cell_click.oncoprint', eventData(d,i,this));
+		}).on('mouseenter', function(d,i) {
+			$(self).trigger('cell_mouseenter.oncoprint', eventData(d,i,this));
+		}).on('mouseleave', function(d,i) {
+			$(self).trigger('cell_mouseleave.oncoprint', eventData(d,i,this));
+		});
+	};
+
+	self.useTemplate = function(templName, params) {
+		// TODO
+		// (1) make sure these are the params you want to pass in
+		// (1a) Question: for genetic alteration, should we pass in design configurations like color?
+		// (2) implement
+
+
+		// DEF: data accessor = string: datum member which holds the category
+		//			| function: takes in datum, outputs value
+		if (templName === 'categorical_color') {
+			// params: - map from category to color
+			//	      - data accessor
+
+		} else if (templName === 'continuous_color') {
+			// params: - data accessor
+			//	      - endpoints of the value range
+			//               - endpoints of the gradient (in same order)
+
+		} else if (templName === 'heat_map') {
+			// params: - data accessor
+			//	      - endpoints of the value range
+
+		} else if (templName === 'bar_chart') {
+			// params: - data accessor
+			//	      - endpoints of the value range
+			//	      - color: string or function of datum
+
+		} else if (templName === 'genetic_alteration') {
+			// any params?
+		}
+	};
+	self.bindEvents = function(track) {
+		$(track).on('sort.oncoprint set_cell_width.oncoprint set_cell_padding.oncoprint', function() {
+			self.updateCells();
+		});
```

Comment:
add binding to set_cell_width and set_cell_padding to increase the width of the svg